client/my-sites/plugins/use-plugins/index.ts

```diff
@@ -80,7 +80,7 @@ const usePlugins = ( {
 		search,
 		tag,
 		{
-			enabled: ! WPCOM_CATEGORIES_BLOCKLIST.includes( category || '' ) && wpcomEnabled,
+			enabled: ! WPCOM_CATEGORIES_BLOCKLIST.includes( category || '' ) && wpcomEnabled && ! search,
```

Comment:
This is the only change related to avoid triggering the request for searches 😅

Reply:
When clicking through categories, the request is still made:
<img width="387" alt="CleanShot 2023-09-08 at 13 33 34@2x" src="https://github.com/Automattic/wp-calypso/assets/11555574/c86ee36c-0989-4ed0-a9be-6938e552ba9e">

However, these results aren't used, because we use the ES results here:
https://github.com/Automattic/wp-calypso/blob/38f28aea3a6d331fbac0e9fcbbaa5d77c6156045/client/my-sites/plugins/use-plugins/index.ts#L92-L114

I'd update this guard to only fetch this with the `paid` category:
```suggestion
			enabled: category === 'paid' && wpcomEnabled && ! search,
```

It's worth considering to also remove the `WPCOM_CATEGORIES_BLOCKLIST` constant. It was useful while we're using this endpoint for the majority of the calls, but now it's a bit misleading IMO.